feat: add `math/base/special/cosf`

anandkaranubc opened the pull request:
Progresses https://github.com/stdlib-js/stdlib/issues/649

## Description

> What is the purpose of this pull request?

This pull request:

-  Adds the complete implementation for `math/base/special/cosf`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   Progresses https://github.com/stdlib-js/stdlib/issues/649

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

anandkaranubc committed:
feat: add `math/base/special/cosf`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
Update package.json

Signed-off-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/cosf/README.md:
```diff
@@ -0,0 +1,187 @@
+// returns 1.0f
+
+out = stdlib_base_cosf( 3.141592653589793f / 4.0f );
+// returns ~-0.707f
```
Per L42, 

```suggestion
// returns ~0.707f
```

anandkaranubc replied:
Thanks for catching that. My bad. This was also present in `cos` and it has been corrected via https://github.com/stdlib-js/stdlib/commit/37d443898b8068b4f2003b839ec29e4afc72fdbe

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/cosf/manifest.json:
```diff
@@ -0,0 +1,90 @@
+      "dependencies": [
+        "@stdlib/math/base/napi/unary",
+        "@stdlib/number/float32/base/to-word",
+        "@stdlib/constants/float64/high-word-abs-mask",
```
`@stdlib/constants/float64/high-word-abs-mask` is unused dependency here and below.

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/cosf/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,135 @@
+	int i;
+
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = ( 20.0f*rand_float() ) - 10.0f;
```
This applies here and elsewhere: For the benchmarks in the statistical distribution packages, we defined a function

```c
/**
* Generates a random number on the interval [min,max).
*
* @param min    minimum value (inclusive)
* @param max    maximum value (exclusive)
* @return       random number
*/
static double random_uniform( const double min, const double max ) {
	double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
	return min + ( v*(max-min) );
}
```

to directly draw from the desires uniform distribution instead of inline calculations such as this. It would be good to do the same in this package's C benchmark files (and `sinf` and others).

anandkaranubc replied:
Done!

anandkaranubc committed:
Merge branch 'develop' into feat/cosf

anandkaranubc committed:
chore: apply suggestions from review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter reviewed (approved):
Looks great; thank you, @anandkaranubc!
